File: apps/android/app/src/main/java/com/warehouseos/operator/data/remote/dto/ProductRequestDto.kt
package com.warehouseos.operator.data.remote.dto

import kotlinx.serialization.Serializable

/** POST /product-requests — worker asks to add a product not in the catalog. */
@Serializable
data class CreateProductRequestBody(
    val name: String,
    val brandName: String? = null,
    val categoryId: String? = null,
    val vehicles: List<String> = emptyList(),
    val quantity: Int = 1,
    val unit: String = "عدد",
    val notes: String? = null,
    val voiceText: String? = null,
    val locationBarcode: String? = null,
    val sessionId: String? = null,
)

@Serializable
data class ProductRequestResult(
    val id: String,
    val status: String = "PENDING",
)

File: apps/api/src/product-requests/dto/product-request.dto.ts
import { Type } from 'class-transformer';
import {
  IsArray,
  IsInt,
  IsOptional,
  IsString,
  MaxLength,
  Min,
} from 'class-validator';

/** Worker-submitted request to add a product not yet in the catalog. */
export class CreateProductRequestDto {
  @IsString()
  @MaxLength(200)
  name: string;

  @IsOptional()
  @IsString()
  @MaxLength(120)
  brandName?: string;

  @IsOptional()
  @IsString()
  categoryId?: string;

  @IsOptional()
  @IsArray()
  @IsString({ each: true })
  vehicles?: string[]; // compatible vehicle names (multiple)

  @IsOptional()
  @IsInt()
  @Min(1)
  quantity?: number;

  @IsOptional()
  @IsString()
  @MaxLength(40)
  unit?: string;

  @IsOptional()
  @IsString()
  @MaxLength(500)
  notes?: string;

  @IsOptional()
  @IsString()
  voiceText?: string;

  @IsOptional()
  @IsString()
  locationBarcode?: string;

  @IsOptional()
  @IsString()
  sessionId?: string;
}

/** Manager edits applied at approval time (all optional — corrects the request). */
export class ApproveProductRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(200)
  name?: string;

  @IsOptional()
  @IsString()
  @MaxLength(120)
  brandName?: string;

  @IsOptional()
  @IsString()
  categoryId?: string;

  @IsOptional()
  @IsArray()
  @IsString({ each: true })
  vehicles?: string[];

  @IsOptional()
  @IsInt()
  @Min(1)
  quantity?: number;

  @IsOptional()
  @IsString()
  @MaxLength(40)
  unit?: string;
}

export class RejectProductRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(500)
  reviewNote?: string;
}
